lib/matplotlib/dates.py

```diff
@@ -1241,10 +1241,7 @@ def __init__(self, bymonthday=None, interval=1, tz=None):
 
         Default is to tick every day of the month: ``bymonthday=range(1,32)``
         """
-        if interval < 1:
-            raise ValueError("The interval parameter must be an integer "
-                             "greater than or equal to one.")
-        if not isinstance(interval, int):
+        if interval < 1 or not isinstance(interval, int):
             raise ValueError("The interval parameter must be an integer "
                              "greater than or equal to one.")
```

Comment:
Ah I realize now that is what story645 was trying to explain. I just missed it. Code is clear as day. Thanks guys :+1: